### PR TITLE
160 fix all warnings during the build

### LINKED ISSUE #160: Fix all warnings during the build
Either fix, or at least clippy-annotate all warnings during the build, so that a default build has zero warnings.

### PR DESCRIPTION
**Pull request by zupzup**
This implements #160 and fixes all warnings and clippy warnings.

The changes were all done in a way so there is no logic change.

### CODE REVIEW

File: src/external/bitcoin.rs [hunk 1]
```diff
@@ -61,68 +45,32 @@ impl AddressInfo {
...
-pub async fn get_transactions_mainnet(address: String) -> Transactions {
-    let request_url = format!(
-        "https://blockstream.info/api/address/{address}/txs",
-        address = address
-    );
+pub async fn get_transactions(address: String) -> Transactions {
+    let request_url = match USEDNET {
+        Network::Bitcoin => {
+            format!(
+                "https://blockstream.info/testnet/api/address/{address}/txs",
```
Comment (mtbitcr) [inline src/external/bitcoin.rs:63]: here no testnet

File: src/external/bitcoin.rs [hunk 2]
```diff
@@ -61,68 +45,32 @@ impl AddressInfo {
...
+                address = address
+            )
+        }
+        _ => {
+            format!(
+                "https://blockstream.info/api/address/{address}/txs",
```
Comment (mtbitcr) [inline src/external/bitcoin.rs:69]: here testnet

File: src/external/bitcoin.rs [hunk 3]
```diff
@@ -139,21 +87,12 @@ impl Txid {
...
-
-    height
-}
-
-pub async fn get_mainnet_last_block_height() -> u64 {
-    let request_url = "https://blockstream.info/api/blocks/tip/height".to_string();
-    let height: u64 = reqwest::get(&request_url)
+pub async fn get_last_block_height() -> u64 {
+    let request_url = match USEDNET {
+        Network::Bitcoin => "https://blockstream.info/testnet/api/blocks/tip/height",
```
Comment (mtbitcr) [inline src/external/bitcoin.rs:92]: here no testnet

**Review by mtbitcr — CHANGES_REQUESTED** [review]
All good, only 3 minor comments

File: src/external/bitcoin.rs (hunk 3 above)
Comment (zupzup) [reply src/external/bitcoin.rs:92]: uh, copy/paste error - nice catch, thanks! :+1:

**Review by mtbitcr — APPROVED** [review]

**Pull request merged**